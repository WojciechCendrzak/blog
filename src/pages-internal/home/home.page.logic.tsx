import { getPostDescriptions } from '../post/post.api';
import { GetStaticProps } from 'next';
import { HomeProps } from './home.page';

export const getStaticProps: GetStaticProps<HomeProps> = async () => {
  const postDescriptions = getPostDescriptions();
  return {
    props: {
      postDescriptions,
    },
  };
};
