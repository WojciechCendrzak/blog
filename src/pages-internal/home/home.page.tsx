import Head from 'next/head';
import { Layout } from '../../components/layout';
import Link from 'next/link';
import { Date } from '../../components/date';
import React from 'react';
import { PostDescription } from '../post/post.model';
import { translate, translationKeys } from '../../logic/translations/translation.service';

export interface HomeProps {
  postDescriptions: PostDescription[];
}

export const HomePage: React.FC<HomeProps> = ({ postDescriptions }) => {
  return (
    <Layout>
      <Head>
        <title>{translate(translationKeys.pages.home.title)}</title>
      </Head>
      <section>
        <h2>Blog</h2>
        <ul>
          {postDescriptions.map(({ id, date, title }) => (
            <li key={id}>
              <Link href={`/post/${id}`}>
                <a>{title}</a>
              </Link>
              <br />
              {date && (
                <small>
                  <Date date={date} />
                </small>
              )}
            </li>
          ))}
        </ul>
      </section>
    </Layout>
  );
};
