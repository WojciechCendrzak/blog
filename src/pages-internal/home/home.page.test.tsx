import renderer from 'react-test-renderer';
import { HomePage } from './home.page';

describe('HomePage', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<HomePage postDescriptions={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
