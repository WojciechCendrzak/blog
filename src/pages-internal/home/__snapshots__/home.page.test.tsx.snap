// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match the snapshot 1`] = `
<div
  className="sc-bdnxRM DVYBs"
>
  <header>
    <h1>
      [Your Name]
       
    </h1>
  </header>
  <main>
    <section>
      <p>
        About me
      </p>
      <p>
        I'am frontend developer
      </p>
    </section>
    <section>
      <h2>
        Blog
      </h2>
      <ul />
    </section>
  </main>
</div>
`;
